### ISSUE TITLE
Users can sniff server-client handshake and get sensible info that should be unknown to their clients

### ISSUE THREAD

**Issue by Oberlus** [freeorion/freeorion#2964]
Bug Report
==========

Environment
-----------
* **FreeOrion Version**: 9eb021f
  * Compiled from source

Description
-----------
Server sends unnecessary info to clients, which allows the users to sniff the packets and deduce information that should be unknown to them or to their clients.
This includes items in the production and research queues. 
The following is an extract of a conversation in the freeorion XMPP chatroom in which user @l29ah revealed such kind of information that should not be possible for him to know.

> (17:57:48) L29Ah: achtung, danyspin97 builds an orbital incubator
> (17:58:58) L29Ah: and O01eg has interstellar logistics research enqueued
> (18:00:16) L29Ah: ThinkSome: watch out for Oberlus, he enqueued lifecycle manipulation
> (18:00:33) L29Ah: and has a sexbomb
> (18:15:29) Querens: L29Ah: describe sexbomb
> (18:16:33) L29Ah: Querens: i can't 
> (18:17:38) Querens: L29Ah: why?
> (18:18:32) L29Ah: dunno how to resolve design ids
> (18:18:48) L29Ah: probably they're private to the empire
> (18:19:07) L29Ah: or i need a separate call for that
> (18:21:52) Querens: I thought you actually see it
> (18:22:59) Voker57: hax
> (18:30:54) Oberlus: L29Ah: so no scout? Would be good to know what you are doing to get that info so that it can be fixed
> (18:31:35) L29Ah: i'm just looking at the freeorion handshake with a sniffer
> (18:31:54) L29Ah: it sends a megabyte worth of xml
> (18:32:04) L29Ah: with various universe info
> (18:36:26) Oberlus: O01eg: I think it's a bit buggy that empire A gets info from empire B that was unknown to it. Because of the obvious hacks and because of unnecessary bandwidth comsumption
> (18:37:35) Oberlus: L29Ah: did you get this by connecting with user "L29Ah" or trying other players' names?
> (18:42:35) O01eg: Oberlus: it's a buggy but someone need to implement server-side calculation for ship's meters.
> (18:43:24) Oberlus: You mean calculations are done in clients?
> (18:43:53) Oberlus: How come then that having the wrong default folder doesn't fool the server?
> (18:44:30) Oberlus: I mean, I can connect with different FOCS files, server says checksum is bad, and I can play
> (18:44:51) Oberlus: Does it mean I could get techs faster by changing the research cost values?
> (18:46:45) O01eg: Oberlus: clients calculates what you see on ships. Combats are calculated on the server.
> (18:48:19) Oberlus: L29Ah: can you confirm if you see or have seen my sexbomb?
> (19:12:20) L29Ah: Oberlus: i've only seen it in your production queue
> (19:12:45) L29Ah: no idea if i can see the existing ships
> (19:12:53) L29Ah: 1MB of XML is hard to parse by eye
> (19:13:46) Oberlus: I asked because if your ships in game haven't seen the sexbomb, why should your client have to calculate anything about those ships? @ O01eg
> (19:14:00) Oberlus: But if it is just the queue, OH MY
> (19:14:50) O01eg: It doesn't calculate for unseen ships. But it get species info and techs to calculate data
> (19:15:09) Oberlus: So it is buggy as hell
> (19:15:41) Oberlus: Because L29Ah's client is "seen" a ship in my build queue

Expected Result
---------------
Sniffing packets does not reveal any sensible information to the user.

Steps to reproduce
------------------
I don't know. Ask @l29ah

**Comment by Voker57** [comment]
Steps to reproduce: 1. sniff packets
2. read xml

**Comment by Voker57** [comment]
#2991 solves the problem.
